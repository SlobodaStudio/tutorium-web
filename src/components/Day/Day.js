import React from 'react'
import { List, ListItem, ListSubHeader} from 'react-toolbox/lib/list'
import styles from './Day.scss'

function Day({ dateAndEvents: { date, events } }) {
  return (
    <ListItem key={date} className={"day-сontent"} >
      <List>
        <ListSubHeader caption={date} />
        {events.map((event) =>
          <ListItem key={event} caption={event} className={styles.event} />
        )}
      </List>
    </ListItem>
  );
};

export default Day
