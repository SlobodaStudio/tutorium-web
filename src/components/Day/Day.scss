@import '../../variables';

.event{
  border-bottom: 1px solid $light-grey;
}
