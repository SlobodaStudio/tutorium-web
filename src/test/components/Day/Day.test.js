import React from 'react';
import ReactDOM from 'react-dom';
import { List, ListItem, ListSubHeader} from 'react-toolbox/lib/list'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Day from '../../../components/Day/Day'

const rootEl = document.createElement('div');
const date = {
                date: "March, 10",
                events: [
                  "Machine learning practice"
                ]
              };

test('can be rendered', () => {
  expect(() => {
    ReactDOM.render(
      <div>
        <Day key={date.date} dateAndEvents={date}/>
      </div>,
      rootEl
    );
  }).not.toThrow();
});

test('renders `ListItem`s', () => {
  const wrapper = shallow(<Day key={date.date} dateAndEvents={date} />);
  expect(wrapper.find('.day-сontent').length).toBe(1);
});

test('Day renders correctly', () => {
  const tree = renderer.create(
    <Day key={date.date} dateAndEvents={date}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
