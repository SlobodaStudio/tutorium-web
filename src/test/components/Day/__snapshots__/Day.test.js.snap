exports[`test Day renders correctly 1`] = `
<li
  className="undefined day-сontent"
  onClick={[Function]}
  onMouseDown={[Function]}
  onTouchStart={[Function]}>
  <span
    className="">
    <span
      className={undefined}>
      <span
        className={undefined}
        onClick={undefined}
        onMouseDown={undefined}>
        <ul
          className=""
          data-react-toolbox="list">
          <h5
            className="">
            March, 10
          </h5>
          <li
            className="undefined "
            onClick={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}>
            <span
              className="">
              <span
                className="">
                <span
                  className="undefined"
                  data-react-toolbox="list-item-text">
                  Machine learning practice
                </span>
              </span>
            </span>
          </li>
        </ul>
      </span>
    </span>
    <span
      className="" />
  </span>
</li>
`;
